live-event-app/live-event-app/RootViewController.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  RootViewController.swift
+//
+//  PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+//  Copyright © 2021 PubNub Inc.
+//  https://www.pubnub.com/
+//  https://www.pubnub.com/terms
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import PubNubChat
+import PubNubChatComponents
+
+class RootViewController: UIViewController {
+  
+  private let provider: PubNubChatProvider
+  private let channelId: String
+  
+  private var portraitConstraints: [NSLayoutConstraint] = []
+  private var landscapeConstraints: [NSLayoutConstraint] = []
+  private weak var scene: UIWindowScene?
+  
+  init(
+    channelId: String,
+    provider: PubNubChatProvider,
+    scene: UIWindowScene
+  ) {
+    self.channelId = channelId
+    self.provider = provider
+    self.scene = scene
+    
+    super.init(nibName: nil, bundle: nil)
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+    
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    guard let scene = scene else {
+      return
+    }
+    // Creates the default MessageList component
+    guard let messageListViewModel = try? provider.messageListComponentViewModel(pubnubChannelId: channelId) else {
```

Comment:
I think that it is better to handle thrown error and use information it is giving

Reply:
MessageList component creation fails only in case of missing required data, so I can change the message text:

`Missing required data. Make sure the channel object and current user are stored locally`